tools/link.py

```diff
@@ -1279,7 +1279,13 @@ def phase_linker_setup(options, state, newargs):
     if sym in settings.EXPORTED_RUNTIME_METHODS:
       settings.REQUIRED_EXPORTS.append(sym)
 
-  settings.REQUIRED_EXPORTS += ['stackSave', 'stackRestore', 'stackAlloc']
+  if settings.MAIN_READS_PARAMS and not settings.STANDALONE_WASM:
```

Comment:
Don't we still need this in standalone wasm mode? We still emit the JS in that case even if the wasm is meant to be standalone, and that JS uses `stackAlloc`. But perhaps we can move the arguments logic into the entry function somehow?

Reply:
In standalone wasm mode `callMain` calls `_start` which takes zero argument, and we don't call stackAlloc.

Reply:
Thanks, that does sound familiar. Then can we ifdef around the `stackAlloc`-using code then? I don't care as a code size optimization (as it's for standalone mode) but closure and other tools might error if we have a call but no function to call.

lgtm though, again, as that's an issue for JS in standalone mode.

Reply:
There is already an `#ifdef` around the use of `staticAlloc` in `callMain`.  See postable.js line 66

Reply:
Ah, you're very right, I was misreading that code. Sorry!